personality/wn/machine-features.pan

```diff
@@ -9,6 +9,26 @@ variable MACHINE_FEATURES_LIST ?= list(
     'total_cpu',
 );
 
+variable MACHINE_FEATURES_SCRIPT ?= 'mjf';
+variable MACHINE_FEATURES_STARTUP = '/etc/init.d/' + MACHINE_FEATURES_SCRIPT;
+'/software/components/filecopy/services' = {
+    SELF[escape(MACHINE_FEATURES_STARTUP)] = dict(
+        'config', file_contents('personality/wn/templ/mjf.init'),
+        'perms', '0755',
+        'owner', 'root',
+        'restart', MACHINE_FEATURES_STARTUP
+    );
+    SELF;
+};
+
+'/software/components/chkconfig' = {
+    if ( index('filecopy', SELF['dependencies']['pre']) < 0 ) {
+        SELF['dependencies']['pre'][length(SELF['dependencies']['pre'])] = 'filecopy';
```

Comment:
Would this be easier to understand?
```pan
SELF['dependencies']['pre'] = append('filecopy');
```

Or you could skip the check and just do:
```pan
prefix '/software/components/chkconfig';
'dependencies/pre' ?= list();
'dependencies/pre' = merge(SELF, list('filecopy'));
'service' ?= dict();
'service' = merge(SELF, dict(
    MACHINE_FEATURES_SCRIPT, 'on',
));
```

Reply:
The first option gives the following error:
```
referenced variable of type nlist is not a list
```
I have other error with the second proposition (I prefer the first).